opentelemetry-otlp/src/span.rs

```diff
@@ -118,6 +237,60 @@ impl Debug for Exporter {
 
 impl Exporter {
     /// Builds a new span exporter with the given configuration
+    #[cfg(feature = "tonic")]
+    pub fn new(
+        config: ExporterConfig,
+    ) -> Result<Self, Box<dyn std::error::Error + Send + Sync + 'static>> {
```

Comment:
Maybe return `Result<Self, crate::Error>` so that we can make sure `crate::Error` covers all possible errors in this crate.

Reply:
Added additional error types to `crate::Error`. Enum names adopted from `tonic` https://github.com/hyperium/tonic/blob/0c69c378fd85d754f045791a5d51b29831e4c728/tonic/src/transport/error.rs#L22-L23